charms/pipelines-persistence/reactive/pipelines_persistence.py

```diff
@@ -15,13 +15,15 @@ def update_image():
     clear_flag('charm.pipelines-persistence.started')
 
 
-@when('layer.docker-resource.oci-image.available')
+@when('layer.docker-resource.oci-image.available', 'pipelines-api.available')
 @when_not('charm.pipelines-persistence.started')
-def start_charm():
+def start_charm(api):
     layer.status.maintenance('configuring container')
 
     image_info = layer.docker_resource.get_info('oci-image')
 
+    api_host = api.services()[0]['hosts'][0]['hostname']
```

Comment:
Do you need to handle multiple http services with multiple hostnames, or is the 0th host's hostname for the 0th service always what you want?

Also, I dunno how likely it is, but it seems possible you can come out `services` with no `hosts`:

https://github.com/juju-solutions/interface-http/blob/master/requires.py#L59

May want to iter over those and return early if you don't have a usable `hostname` for any of the service hosts.

Reply:
The service code here only handles a single API endpoint, so multiple services returned from `.services()` shouldn't be an issue, and IIUC, unless I've miscoded something, we should always have at least one service. I could definitely be using it wrong, though.